olp-cpp-sdk-dataservice-write/include/olp/dataservice/write/FlushEventListener.h

```diff
@@ -88,50 +67,19 @@ class FlushEventListener {
 template <typename FlushResponse>
 class DefaultFlushEventListener : public FlushEventListener<FlushResponse> {
  public:
-  /**
-   Number of flush events attempted.
-
-   @return Number of the flush events attempted
-   */
-  size_t GetNumFlushEventsAttempted() const override {
-    return num_flush_events_attempted_;
-  }
-  /**
-   Get the number of flush events failed.
-
-   @return number of failues.
-   */
-  size_t GetNumFlushEventsFailed() const override {
-    return num_flush_events_failed_;
-  }
-  /**
-   Total number of flush events.
-
-   @return Total number of flush events.
-   */
-  size_t GetNumFlushEvents() const override { return num_flush_events_; }
-  /**
-   Number of flushed requests.
-
-   @return Number of flushed requests
-   */
-  size_t GetNumFlushedRequests() const override {
-    return total_flush_requests_;
-  }
-  /**
-   Number of requests that were failed to be flushed.
-
-   @return Number of failed requests
-   */
-  size_t GetNumFlushedRequestsFailed() const override {
-    return num_flush_requests_failed_;
-  }
-
   /**
    Notify the start of the flush event.
 
    */
-  void NotifyFlushEventStarted() override { ++num_flush_events_attempted_; }
+  void NotifyFlushEventStarted() override {
+    FlushMetrics metrics;
+    {
+      std::lock_guard<std::mutex> locker(mutex_);
+      ++metrics_.num_attempted_flush_events;
+      metrics = metrics_;
+    }
+    NotifyFlushMetricsHasChanged(std::move(metrics));
```

Comment:
so we're locking the mutex here and once again in the notify?

Reply:
No, it's up to the user to decide where they want to lock smith in the `NotifyFlushMetricsHasChanged`. Or which `Notify` method do you mean?